modules/api/src/main/java/org/apache/ignite/tx/IncompatibleSchemaException.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.tx;
+
+import java.util.UUID;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Thrown when, during an attempt to execute a transactional operation, it turns out that the operation cannot be executed
+ * because an incompatible schema change has happened.
+ */
+public class IncompatibleSchemaException extends TransactionException {
```

Comment:
Not sure if we should inherit from `TransactionException` here. 

All table operations are transactional in Ignite 3, so technically any schema incompatibility issue is related to some transaction, but it does not mean we should inherit all table update exceptions from `TransactionException`.

Reply:
Made `IncompatibleSchemaException` extend `IgniteException` and moved it to `org.apache.ignite`.

Reply:
After thinking about this a bit more, I've reverted this change, so it's again a `TransactionException` defined in `org.apache.ignite.tx`. The reasoning is that this type of exceptions can only be thrown by transactional operations (like `put()` or `commit()`), and yet not all API operations in Ignite 3 are transactional: for example, there is Compute API which isn't.